File: _bak/version/cpy0i_v0.1/function.c
#include "function.h"

#include "util.h"

#include <stdlib.h>

PyValue *py_make_function(const char *name, char **params, int param_count, AstNode *body, PyEnv *closure) {
    PyFunction *fn = (PyFunction *)calloc(1, sizeof(PyFunction));
    if (!fn) die("out of memory");
    fn->name = xstrdup(name);
    fn->params = params;
    fn->param_count = param_count;
    fn->body = body;
    fn->closure = closure;
    return py_new_function(fn);
}

File: _bak/cpy0/rust/src/function.rs
use crate::ast_nodes::AstNode;
use crate::value::{PyEnv, PyFunction, PyValue};

pub fn py_make_function(
    name: String,
    params: Vec<String>,
    param_count: usize,
    body: AstNode,
    closure: *const PyEnv,
) -> PyValue {
    PyValue::new_function(PyFunction {
        name,
        params,
        param_count,
        body,
        closure,
    })
}